File: siki-front/hku-alpha/src/api/voucher.ts
import request from '@/utils/request'
import type { PageQuery, PageResult } from '@/types'

// 后端 DTO 和 Entity 的简化 TypeScript 接口
export interface VoucherAddDTO {
  title: string
  subTitle?: string
  rules?: string
  actualValue: number // 分
  payValue: number // 分
  beginTime: string // LocalDateTime 字符串
  endTime: string // LocalDateTime 字符串
  stock: number
}

export interface Voucher extends VoucherAddDTO {
  id: number
  status: number // 1-上架 2-下架 3-过期
  createTime: string // LocalDateTime 字符串
}

// 新增代金券
export const addVoucherAPI = (data: VoucherAddDTO) => {
  return request({
    url: '/voucher',
    method: 'post',
    data
  })
}

// 分页查询代金券
export const getVoucherPageListAPI = (params: PageQuery) => {
  return request<PageResult<Voucher>>({
    url: '/voucher/page',
    method: 'get',
    params
  })
}

// 根据ID查询代金券 (如果需要编辑功能)
export const getVoucherByIdAPI = (id: number) => {
  return request<Voucher>({
    url: `/voucher/${id}`,
    method: 'get'
  })
}

// 更新代金券状态 (上架/下架)
export const updateVoucherStatusAPI = (id: number, status: number) => {
  return request({
    url: `/voucher/status/${id}/${status}`,
    method: 'put'
  })
}

// 更新代金券信息 (如果需要编辑功能，且后端接口与新增不同)
// export const updateVoucherAPI = (data: Voucher) => {
//   return request({
//     url: '/admin/voucher',
//     method: 'put',
//     data
//   })
// }

export const deleteVoucherAPI = (id: number) => {
  return request({
    url: `/voucher/${id}`,
    method: 'delete'
  })
}

File: siki-pojo/src/main/java/com/siki/entity/Voucher.java
package com.siki.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Voucher implements Serializable{
    private static final long serialVersionUID = 1L;

    private Long id;

    //代金券名称
    private String title;

    //代金券描述
    private String subTitle;

    //代金券使用说明
    private String rules;

    //代金券金额
    private Long actualValue;

    //支付金额
    private Long payValue;

    //代金券开始时间
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime beginTime;

    //代金券结束时间
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime endTime;

    //剩余库存
    private Integer stock;

    //优惠券状态 1-上架 2-下架 3-过期
    private Integer status;

    //创建时间
    private LocalDateTime createTime;

    //更新时间
    private LocalDateTime updateTime;
}
